ecl2df/compdat.py

```diff
@@ -955,8 +955,12 @@ def fill_parser(parser: argparse.ArgumentParser) -> argparse.ArgumentParser:
         "-o",
         "--output",
         type=str,
-        help="Name of output csv file.",
-        default="compdat.csv",
+        help=(
+            "Override name of output csv file.\n"
+            + "Otherwise name is derived from datafile and datatype.\n"
+            + "Use '-' for stdout."
+        ),
+        default=None,
```

Comment:
This changes the default behavior? Will this create problems for existing users (who will expect a file called `compdat.csv` still?)

Reply:
Is it better to explicitly trigger the auto detection (at least for a period of time), and look into changing the defaults later when we have confirmed that it is stable? Could be quite a lot of unexpected effects here on existing usage. Perhaps using `name=auto` or similar. That way, the feature is included and available, but it still gives a chance to learn how it works "in the wild".

Reply:
Well, you of course have a point when it comes to existing users, but my feeling is that this is rarely executed with only the default. Because this will give the user a file names compdat.csv in the current folder

 What is implemented now is parallel to this but makes a file called <name of datafile>--compdat.csv in stead. Still in the current folder.. 

If they use --output the behaviour will be identical to before. But what I am more worried about is if people want autodetection, but put also to put it in a different folder, that will not work. So I have considered to make functionality that check if output is a directory, and if so prepend this to output name. Remember that most of the time this will be run through ert as a forward model called ECL2CSV, here I see that you need to put in <output>. About the autodetection as a specific option, to me this is basically what None does. But it would be good to get feedback from people that use this more regularly..